### DISCUSSION TITLE
Sonoff ZBBridge fails to connect with FDI port

### DISCUSSION THREAD

**Discussion by SmartKobol** [arendst/Tasmota#12006] [Q&A] [answered]
Dear Folks,
I am tasmotizing my Sonoff products, so far without any problem. FTDI device (FT232BL) has hooked up with the Win7 and worked well and Tasmotizer1.2 has been applied for flashing. I have tried the same procedure with a ZBBridge (all connection as written) BUT when I plug the FTDI/USB it does not connect as a new USB serial port. Led is green but that is all, no familiar sound indicated plugging new device . What is weird and unusual that the TL and RL indicator leds are flashing faintly showing some data exchange? After all ZB connections terminated, powered off and ZB factory firmware backloaded nothing has changed. The device is waiting for pairing (blue 2-1 flashing) with WIFI when plug in. The situation is rather annoying as I am failing to launch Tasmotizer (or any other flashing apps). Have you any explanation and tip/solution to get rid of this failure?

**Comment by sfromis** [comment, marked as answer]
As the serial port is being used to communicate with the EZSP chip, it is not available to other Tasmota usage like serial console after flashing. You have to do everything via Wifi. Note that those FT232BL devices are *not recommended* as they have too weak power for much. You may be lucky to flash without error, but you cannot have it power the bridge while Wifi is needed.

**Reply by SmartKobol** [reply]
Dear sformis, Thank you for your reply. I have the problem even at the first FTDI  flashing from factory firmware to 9.4.0  (as Roberto Jorge, Home Sight, DigiBlur etc) suggests on the Youtube.  Then and only then I can upgrade the device (EZSP) with OTA through the Tasmota platform. Am I right, or can I do the first step via OTA avoiding FTDI mess? I accept your opinion about FT232BL, but as I see this is the most widely USB used in flashing. True that VCC is only 3.1 V ,but what else you can suggest?

**Reply by Jason2866** [reply]
As written in Tasmota [docs](https://tasmota.github.io/docs/Getting-Started/#serial-to-usb-adapter) we suggest the golden ch340

**Reply by sfromis** [reply]
My statement about FT232BL was not "opinion" but plain fact. It can only deliver about 50mA of 3.3V power, which is just too low for full operation of even the ESP8266 chip itself. That adapter is just not designed to act as a power supply. The recommended "golden CH340G" has an extra voltage regulator, enabling it to supply about 150mA of 3.3V power. Much better. 

Unfortunately, a lot of people on YouTube use that weak adapter, but we have seen many troubles go away after advising people to use a good adapter instead. The FT232BL is not bad as such, expecting the adapter to act as a power supply is just a function many adapters are not meant for.

You cannot use OTA for the initial flashing of Tasmota, you need an adapter. If you have a NodeMCU around, that can actually be used as an even better adapter with 500mA of 3.3V power, if you connect RST/GND to disable the onboard ESP8266, and connect TX/RX straight instead of crossed.

If flashing as such is successful, you should be able to operate the bridge after disconnecting the adapter, and powering the bridge directly via Micro-USB, assuming that your power source is good.

**Reply by SmartKobol** [reply]
> 
> 
> As written in Tasmota [docs](https://tasmota.github.io/docs/Getting-Started/#serial-to-usb-adapter) we suggest the golden ch340

Thank you for you feedback!

**Reply by SmartKobol** [reply]
Thank you for your clarification and technical reference!  I will purchase the "golden device" and start from the beginning.

**Reply by SmartKobol** [reply]
Dear All, Unfortunately I still have the tasmotizing problem. However, I use a new FTDI (CH340G-M) as suggested still sticking at the very first phase. After connecting the USB-serial port, checking that it is the active and changing bit/sec 115200 I got "Timed out waiting packing header" message, shortly after tasmotizing pop-up appears. It is the same for tasmota_zbridge 9.4.0 ans 9.4.0.3, too. Where I did I go wrong?

**Reply by sfromis** [reply]
I do not see you mention having IO0 grounded while applying power, which is a requirement for getting into flashing mode. And the symptom could match that cause.

**Reply by SmartKobol** [reply]
Dear sfromis, IO0 and GND are connected together and goes to the FTDI's GND. I wire the FTDI & ZBB as presented, 5 holes on ZBB connecting to the 4 pins of the FTDI. Pairing button is pressed and then stick plug-in the computer. When I release the button the green led lights a bit stronger. Start Tasmotizer and after 10-15 seconds (new) message appears "Failed to connect to ESP8266: Timed out waiting for packet header". And that is all... Voltage on the pins RX/TX and VCC: 3,6-3,7V, 3,3V: 3,24V 5V: 5,05V. Any tip how can I get rid of this problem. Rather annoying...

**Reply by SmartKobol** [reply]
And one more info: I want to keep the original firmware just in case. Therefore "Save original firmware" has been ticked in the Tasmotizer. However, when I tried to flash the device with the ESPeasy flasher, - without keeping the factory firmware - the result was the same.

**Reply by Jason2866** [reply]
Did you do as described here https://templates.blakadder.com/sonoff_ZBBridge.html
Do not power the bridge via USB for flashing.

**Reply by SmartKobol** [reply]
I did exactly the same procedure, wires connected as blakadder suggests, and microUSB disconnected, only flash wires led to the device. Green light indicates the programming status, and I got the error message...

**Reply by SmartKobol** [reply]
What do you thing, output voltage on the FTDI pins is acceptable?

**Reply by SmartKobol** [reply]
Update: Dear Jason2866 and sfromis and others, A new phenomena appears: flashing starts and erase done. Then MAC address and ESP8266EX written but flashing process terminates at 11% with "Timed out for packet header" message. Very frustrating that after flashing several Sonoff switches with the same FTDI adapters I cannot complete a routine "tasmotizing" for Sonoff Zigbee Bridge now... Weak/inappropriate  adapter or dead flash memory on the chip? I appreciate any suggestion.

**Reply by Jason2866** [reply]
Use the already recommended "golden" CH340. The fact that your adapter worked for other devices does say nothing.

**Reply by SmartKobol** [reply]
Dear Jason2866, Thank you for your reply. Seems than golden ch340 is the solution. I have ordered one, hoping problem will be solved.
